node/src/types/validator_matrix.rs

```diff
@@ -122,17 +130,21 @@ impl ValidatorMatrix {
             .write()
             .expect("poisoned lock on validator matrix");
         let is_new = guard.insert(era_id, validators).is_none();
-        if guard.len() > MAX_VALIDATOR_MATRIX_ENTRIES {
-            // Safe to unwrap because we check above that we have sufficient entries.
-            let median_key = guard
-                .keys()
-                .nth(MAX_VALIDATOR_MATRIX_ENTRIES / 2)
-                .copied()
-                .unwrap();
-            guard.remove(&median_key);
-            return median_key != era_id;
+        if guard.len() <= MAX_VALIDATOR_MATRIX_ENTRIES || self.retrograde_latch.is_none() {
+            return is_new;
+        }
+        let latch_era = self.retrograde_latch.unwrap_or_default(); // none checked above
+        let median_era = guard
+            .keys()
+            .nth(MAX_VALIDATOR_MATRIX_ENTRIES / 2)
+            .copied()
+            .unwrap();
+        if median_era <= latch_era {
+            is_new
+        } else {
+            guard.remove(&median_era);
+            median_era != era_id
```

Comment:
On this case we should probably keep removing entries in a loop until we reach `MAX_VALIDATOR_MATRIX_ENTRIES` entries.